orca-clouddriver/src/main/groovy/com/netflix/spinnaker/orca/clouddriver/tasks/MonitorKatoTask.groovy

```diff
@@ -91,14 +92,18 @@ class MonitorKatoTask implements RetryableTask {
     new DefaultTaskResult(status, outputs)
   }
 
-  private static ExecutionStatus katoStatusToTaskStatus(Task katoTask, boolean katoResultExpected) {
+  private static ExecutionStatus katoStatusToTaskStatus(Task katoTask, boolean katoResultExpected, String katoResultKeyExpected) {
     def katoStatus = katoTask.status
     if (katoStatus.failed) {
       return ExecutionStatus.TERMINAL
     } else if (katoStatus.completed) {
       if (katoResultExpected && !katoTask.resultObjects) {
         return ExecutionStatus.RUNNING
       }
+      if (katoResultExpected && katoTask.resultObjects
+        && katoResultKeyExpected && !katoTask.resultObjects.any { it.containsKey(katoResultKeyExpected) }) {
+        return ExecutionStatus.RUNNING
```

Comment:
Not a huge fan of another bandaid being applied here. 

I'd still really like to know what's going on here ... at the very least see a metric for how frequently this is happening. 

I find it hard to imagine that doing a re-read against clouddriver would return the correct results ... unless I'm missing something `clouddriver` doesn't `.complete()` a task until after `.addResultObjects()` has been called.

Would additional logging in `addResultObjects` (clouddriver) be helpful?